File: app/src/main/java/com/letianpai/robot/control/callback/RobotCommandWordsCallback.java
package com.letianpai.robot.control.callback;


import android.util.Log;

import com.renhejia.robot.commandlib.parser.tips.Tips;

/**
 * Created by liujunbin
 */

public class RobotCommandWordsCallback {

    private RobotCommandWordsUpdateListener mRobotCommandWordsUpdateListener;

    private static class RobotCommandWordsCallbackHolder {
        private static RobotCommandWordsCallback instance = new RobotCommandWordsCallback();
    }

    private RobotCommandWordsCallback() {}

    public static RobotCommandWordsCallback getInstance() {
        return RobotCommandWordsCallbackHolder.instance;
    }

    public interface RobotCommandWordsUpdateListener {
        void showBattery(boolean showBattery);
    }

    public void setRobotCommandWordsUpdateListener(RobotCommandWordsUpdateListener listener) {
        this.mRobotCommandWordsUpdateListener = listener;
    }

    public void showBattery(boolean showBattery) {
        if (mRobotCommandWordsUpdateListener != null) {
            mRobotCommandWordsUpdateListener.showBattery(showBattery);
        }
    }

}

File: app/src/main/java/com/letianpai/robot/control/callback/RobotCommandWordsCallback.kt
package com.letianpai.robot.control.callback


/**
 * Created by liujunbin
 */
class RobotCommandWordsCallback private constructor() {
    private var mRobotCommandWordsUpdateListener: RobotCommandWordsUpdateListener? = null

    private object RobotCommandWordsCallbackHolder {
        val instance: RobotCommandWordsCallback = RobotCommandWordsCallback()
    }

    interface RobotCommandWordsUpdateListener {
        fun showBattery(showBattery: Boolean)
    }

    fun setRobotCommandWordsUpdateListener(listener: RobotCommandWordsUpdateListener?) {
        this.mRobotCommandWordsUpdateListener = listener
    }

    fun showBattery(showBattery: Boolean) {
        if (mRobotCommandWordsUpdateListener != null) {
            mRobotCommandWordsUpdateListener!!.showBattery(showBattery)
        }
    }

    companion object {
        val instance: RobotCommandWordsCallback
            get() = RobotCommandWordsCallbackHolder.instance
    }
}












